simulator/infra/cache/cache_tag_array.cpp

```diff
@@ -167,3 +131,12 @@ uint32 CacheTagArray::write( Addr addr)
 
     return way;
 }
+
+ReplacementModule::ReplacementModule( std::size_t number_of_sets, std::size_t number_of_ways, int replacement_policy)
+{
+    if (replacement_policy == LRU)
```

Comment:
Usually enum is a poor choice for factory, as it is creates artificial binding from implementations to the interface.
The more common thing is to use string.